tensorboard/uploader/exporter_test.py

```diff
@@ -227,6 +229,328 @@ def outdir_files():
+        self.assertEqual(next(generator), "123")
+        # Check the blob_sequences.json file.
+        with open(
+            os.path.join(outdir, "experiment_123", "blob_sequences.json")
+        ) as infile:
+            jsons = [json.loads(line) for line in infile]
+        self.assertLen(jsons, 1)
+
+        datum = jsons[0]
+        self.assertEqual(datum.pop("run"), "train")
+        self.assertEqual(datum.pop("tag"), "__default_graph__")
+        summary_metadata = summary_pb2.SummaryMetadata.FromString(
+            base64.b64decode(datum.pop("summary_metadata"))
+        )
+        expected_summary_metadata = test_util.scalar_metadata(
+            "123:__default_graph__"
+        )
+        self.assertEqual(summary_metadata, expected_summary_metadata)
+        points = datum.pop("points")
+        self.assertEqual(points["steps"], [0])
+        self.assertEqual(points["wall_times"], [1571084520.862939144])
+        # `None` represents the blob that is unfinished and hence is not
+        # exported.
+        self.assertEqual(points["blob_file_paths"], [[None]])
+
+    def test_e2e_success_case_with_blob_streaming_error(self):
+        mock_api_client = self._create_mock_api_client()
+        mock_api_client.StreamExperiments.return_value = iter(
+            [_make_experiments_response(["789"])]
+        )
+
+        def stream_experiments(request, **kwargs):
+            del request  # unused
+            self.assertEqual(kwargs["metadata"], grpc_util.version_metadata())
+
+            response = export_service_pb2.StreamExperimentsResponse()
+            response.experiments.add(experiment_id="123")
+            yield response
+
+        def stream_experiment_data(request, **kwargs):
+            self.assertEqual(kwargs["metadata"], grpc_util.version_metadata())
+
+            run = "train"
+            tag = "__default_graph__"
+            response = export_service_pb2.StreamExperimentDataResponse()
+            response.run_name = run
+            response.tag_name = tag
+            display_name = "%s:%s" % (request.experiment_id, tag)
+            response.tag_metadata.CopyFrom(
+                test_util.scalar_metadata(display_name)
+            )
+            for step in range(1):
+                response.blob_sequences.steps.append(step)
+                response.blob_sequences.wall_times.add(
+                    seconds=1571084520 + step, nanos=862939144
+                )
+                blob_sequence = blob_pb2.BlobSequence()
+                blob = blob_pb2.Blob(
+                    blob_id="%s_blob" % run,
+                    state=blob_pb2.BlobState.BLOB_STATE_CURRENT,
+                )
+                blob_sequence.entries.append(
+                    blob_pb2.BlobSequenceEntry(blob=blob)
+                )
+                response.blob_sequences.values.append(blob_sequence)
+            yield response
+
+        def stream_blob_data(request, **kwargs):
+            raise grpc.RpcError("Error for testing")
```

Comment:
Please use `test_util.grpc_error` to more faithfully match the actual
errors that we see in prod.